src/web/app/pages-instructor/instructor-student-records-page/instructor-student-records-page.component.ts

```diff
@@ -56,10 +75,103 @@ export class InstructorStudentRecordsPageComponent implements OnInit {
+   */
+  loadStudentResults(): void {
+    this.feedbackSessionsService.getFeedbackSessionsForInstructor(this.courseId).subscribe((resp: FeedbackSessions) => {
+      resp.feedbackSessions.forEach((feedbackSession: FeedbackSession) => {
+        this.feedbackSessionsService.getFeedbackSessionsResult({
+          courseId: this.courseId,
+          feedbackSessionName: feedbackSession.feedbackSessionName,
+          groupBySection: this.studentSection,
+          intent: Intent.INSTRUCTOR_RESULT,
+        }).subscribe((results: SessionResults) => {
+          results.questions.forEach((q: QuestionOutput) => {
+            q.allResponses = q.allResponses.filter((response: ResponseOutput) =>
+                response.recipient === this.studentProfile.name ||
+                response.giver === this.studentProfile.name);
+          });
+          const questions: QuestionOutput[] =
+              results.questions.filter((q: QuestionOutput) => q.allResponses.length > 0);
+          this.sessions.push({
+            questions,
+            feedbackSession,
+            isCollapsed: questions.length === 0,
+          });
+          results.questions.forEach((qn: QuestionOutput) => this.preprocessComments(qn.allResponses));
+        });
+      });
+    }, (errorMessageOutput: ErrorMessageOutput) => {
+      this.statusMessageService.showErrorMessage(errorMessageOutput.error.message);
+    });
+  }
+
+  /**
+   * Preprocesses the comments from instructor.
+   *
+   * <p>The instructor comment will be moved to map {@code instructorCommentTableModel}. The original
+   * instructor comments associated with the response will be deleted.
+   */
+  preprocessComments(responses: ResponseOutput[]): void {
+    responses.forEach((response: ResponseOutput) => {
+      this.instructorCommentTableModel[response.responseId] =
+          this.getCommentTableModel(response.instructorComments);
+
+      // clear the original comments for safe as instructorCommentTableModel will become the single point of truth
+      response.instructorComments = [];
+    });
+  }
+
+  /**
+   * Transforms instructor comments to a comment table model.
+   */
+  getCommentTableModel(comments: CommentOutput[]): CommentTableModel {
+    return {
+      commentRows: comments.map((comment: FeedbackResponseComment) => this.getCommentRowModel(comment)),
+      newCommentRow: {
+        commentEditFormModel: {
+          commentText: '',
+          isUsingCustomVisibilities: false,
+          showCommentTo: [],
+          showGiverNameTo: [],
+        },
+
+        isEditing: false,
+      },
+      isAddingNewComment: false,
+    };
+  }
+
+  /**
+   * Transforms a comment to a comment row model.
+   */
+  getCommentRowModel(comment: CommentOutput): CommentRowModel {
+    return {
+      originalComment: comment,
+      timezone: this.sessions[0] != null ? this.sessions[0].feedbackSession.timeZone : '',
```

Comment:
Maybe you want to write a comment here?

Sessions in the same course will always have the same timezone?